src/eckit/types/Time.cc

```diff
@@ -67,60 +98,56 @@ Time::Time(const std::string& s, bool extended) {
+        hh   = sec / 3600;
+        sec -= hh * 3600;
+        mm   = sec / 60;
+        sec -= mm * 60;
+        ss   = sec;
+    } else if (format == TimeFormat::OTHER) {
+        std::smatch m;
+        if (std::regex_match(s, m, hhmmss_)) {
+            for (int i = 1; i < m.size(); i++) {
+                if (m[i].matched) {
+                    switch (i) {
+                        case 1: hh = std::stol(m[i].str()); break;
+                        case 2: mm = std::stol(m[i].str()); break;
+                        case 3:
+                            std::string aux = m[i].str();
+                            aux.erase(0, 1);
+                            ss = std::stol(aux);
+                            break;
                     }
                 }
             }
-            else {
-                if (std::regex_match (s, m, ddhhmmss_)) {
-                    for (int i=1; i<m.size(); i++) {
-                        if (m[i].matched) {
-                            std::string aux = m[i].str();
-                            aux.pop_back();
-                            long t = std::stol(aux);
-                            switch (i) {
-                                case 1: dd = t; break;
-                                case 2: hh = t; break;
-                                case 3: mm = t; break;
-                                case 4: ss = t;
-                            }
-                        }
-                    }
-                    ss += 60 * (mm + 60 * (hh + 24 * dd));
-                    if (s[0] == '-') {
-                        ss = -ss;
+        } else if (std::regex_match(s, m, ddhhmmss_)) {
+            for (int i = 1; i < m.size(); i++) {
+                if (m[i].matched) {
+                    std::string aux = m[i].str();
+                    aux.pop_back();
+                    long t = std::stol(aux);
+                    switch (i) {
+                        case 1: dd = t; break;
+                        case 2: hh = t; break;
+                        case 3: mm = t; break;
+                        case 4: ss = t;
                     }
-                    dd =  ss / 86400;
-                    hh = (ss /  3600) % 24;
-                    mm = (ss /    60) % 60;
-                    ss =  ss          % 60;
-                } else {
-                    std::string msg = "Wrong input for time: ";
-                    msg += s;
-                    throw BadTime(msg);
                 }
             }
+            ss += 60 * (mm + 60 * (hh + 24 * dd));
+            if (s[0] == '-') { ss = -ss; }
+            dd = ss / 86400;
+            hh = (ss / 3600) % 24;
+            mm = (ss / 60) % 60;
+            ss = ss % 60;
+        } else {
+            throw BadTime("Wrong input for time: " + s);
         }
+    } else if (format == TimeFormat::UNKOWN) {
+        throw BadTime("Unkown format for time: " + s);
+    } else {
+        throw SeriousBug("Unhandled time format!");
```

Comment:
Why is there a need for two different exceptions? I understand the intent of those exception types as follows: 
- `BadTime` is best described as a usage error, as in it was called as `Time("ABCD")`
- `SeriousBug` represents an implementation error such as `Time("20:11:24")` not getting parsed properly.

Is that second throw giving any additional benefits?